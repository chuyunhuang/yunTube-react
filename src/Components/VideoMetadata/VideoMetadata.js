import React from 'react';
import { Divider } from 'semantic-ui-react';
import './VideoMetadata.scss';

export function VideoMetadata(props) {
  return (
    <div className="video-metadata">
      <h3>Video Title</h3>
      <div className="video-stats">
        <span>views</span>
      </div>
      <div>Sth</div>
      <Divider />
    </div>
  )
}