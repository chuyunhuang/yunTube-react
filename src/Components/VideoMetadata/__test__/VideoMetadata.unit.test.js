import React from 'react';
import { shallow } from 'enzyme';
import { VideoMetadata } from '../VideoMetadata';

describe('VideoMetadata', () => {
  test('renders VideoMetadata without props', () => {
    const wrapper = shallow(<VideoMetadata />)
    expect(wrapper).toMatchSnapshot();
  });

  test('renders VideoMetadata with viewCount', () => {
    const wrapper = shallow(<VideoMetadata viewCount={25} />)
    expect(wrapper).toMatchSnapshot();
  });
});
