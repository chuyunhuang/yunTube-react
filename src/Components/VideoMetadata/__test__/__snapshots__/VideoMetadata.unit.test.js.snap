// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMetadata renders VideoMetadata 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Video Title
  </h3>
  <div
    className="video-stats"
  >
    <span>
       1,000 views
    </span>
    <div
      className="video-action"
    >
      <Rating
        dislikeCount={1}
        likeCount={20}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelposition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`VideoMetadata renders VideoMetadata with viewCount 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Video Title
  </h3>
  <div
    className="video-stats"
  >
    <span>
       1,000 views
    </span>
    <div
      className="video-action"
    >
      <Rating
        dislikeCount={1}
        likeCount={20}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelposition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`VideoMetadata renders VideoMetadata without props 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Video Title
  </h3>
  <div
    className="video-stats"
  >
    <span>
       1,000 views
    </span>
    <div
      className="video-action"
    >
      <Rating
        dislikeCount={1}
        likeCount={20}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelposition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="add circle"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
